save-backend/src/main/kotlin/com/saveourtool/save/backend/repository/TimestampBasedFileSystemRepository.kt

```diff
@@ -1,89 +1,53 @@
 package com.saveourtool.save.backend.repository
 
-import com.saveourtool.save.backend.configs.ConfigProperties
-import com.saveourtool.save.domain.FileInfo
-import com.saveourtool.save.domain.ImageInfo
-import com.saveourtool.save.domain.ProjectCoordinates
-import com.saveourtool.save.domain.ShortFileInfo
+import com.saveourtool.save.backend.storage.AvatarKey
+import com.saveourtool.save.backend.storage.AvatarStorage
+import com.saveourtool.save.backend.storage.FileStorage
+import com.saveourtool.save.domain.*
 import com.saveourtool.save.utils.AvatarType
+import com.saveourtool.save.utils.toDataBufferFlux
 
 import org.slf4j.LoggerFactory
 import org.springframework.core.io.FileSystemResource
 import org.springframework.http.codec.multipart.FilePart
 import org.springframework.stereotype.Repository
 import reactor.core.publisher.Flux
 import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
+import java.nio.ByteBuffer
 
-import java.nio.file.FileSystemException
-import java.nio.file.Files
 import java.nio.file.Path
-import java.nio.file.Paths
-import java.nio.file.StandardOpenOption.APPEND
-import java.util.stream.Collectors
 import kotlin.io.path.*
 
 /**
  * A repository which gives access to the files in a designated file system location
  */
 @Repository
-class TimestampBasedFileSystemRepository(configProperties: ConfigProperties) {
+class TimestampBasedFileSystemRepository(
```

Comment:
removed all private methods and rewrote public ones. but have to change signature for `getFile` method